doc/akri_connector/connector-metadata-schema.json

```diff
@@ -0,0 +1,305 @@
+    },
+    "description": {
+      "type": "string",
+      "description": "A brief (human-readable) description of the connector."
+    },
+    "version": {
+      "type": "string",
+      "description": "The version of this connector."
+    },
+    "maintainer": {
+      "type": "string",
+      "description": "A name and/or email of the maintainer of this connector."
+    },
+    "vendor": {
+      "type": "string",
+      "description": "A name and/or email of the vendor that distributes this connector."
+    },
+    "imageConfigurationSettings": {
+      "type": "object",
+      "properties": {
+        "containerRegistryName": {
+          "type": "string",
+          "description": "The container registry that holds the image for the connector. For example: contoso-acr.io"
+        },
+        "imageName": {
+          "type": "string",
+          "description": "The docker image name for the connector. For example: MyConnector"
+        },
+        "tag": {
+          "type": "string",
+          "description": "The tag of the docker image for this connector"
+        },
+        "digest": {
+          "type": "string",
+          "description": "The digest of the docker image for this connector"
+        }
+      },
+      "oneOf": [
+        {
+          "required": [
+            "containerRegistryName",
+            "imageName",
+            "tag"
+          ]
+        },
+        {
+          "required": [
+            "containerRegistryName",
+            "imageName",
+            "digest"
+          ]
+        }
+      ]
+    },
+    "helmConfigurationSettings": {
+      "type": "object",
+      "properties": {
+        "containerRegistryName": {
+          "type": "string",
+          "description": "The container registry name that holds the helm chart for the connector. For example: contoso-acr.io"
+        },
+        "helmChartName": {
+          "type": "string",
+          "description": "The helm chart name for the connector. For example: helm/MyConnector"
+        },
+        "tag": {
+          "type": "string",
+          "description": "The tag of the helm chart for this connector"
+        },
+        "digest": {
+          "type": "string",
+          "description": "The digest of the helm chart for this connector"
+        }
+      },
+      "oneOf": [
+        {
+          "required": [
+            "containerRegistryName",
+            "helmChartName",
+            "tag"
+          ]
+        },
+        {
+          "required": [
+            "containerRegistryName",
+            "helmChartName",
+            "digest"
+          ]
+        }
+      ]
+    },
+    "kubernetes": {
+      "type": "object",
+      "description": "The kubernetes-level metadata",
+      "properties": {
+        "minimumKubernetesVersion": {
+          "type": "string",
+          "description": "Minimum required version of kubernetes needed to run this connector"
+        }
+      }
+    },
+    "secrets": {
+      "type": "object",
```

Comment:
@abhipsaMisra @varunpuranik @timtay-microsoft 
1) this has to be an array of secrets
2) i am not sure how this will work end to end:
- the secrets field is at top level, so portal need to handle it
-  there must be a description for the user what the purpose of the secret is
- we should add an optional flag as property as well to ensure the portal allows to not add it
- is the assumption that the user does create a secret with name secretRef ahead of deployment?
- does portal creates the secret during import?
- the connector is using secretAlias and secretKey, so secretRef could even added by the user in the portal. no?

Reply:
I made the secrets an array now and added two new properties alongside "secretAlias": "isOptional" and "secretDescription".